realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -628,7 +628,7 @@ private void completedAsyncRealmObject(QueryUpdateTask.Result result) {
                 } else if (compare > 0) {
                     // the caller has advanced we need to
                     // retry against the current version of the caller if it's still empty
-                    if (RealmObject.isValid(proxy)) { // already completed & has a valid pointer no need to re-run
+                    if (RealmObject.isLoaded(proxy)) { // already completed & has a valid pointer no need to re-run
```

Comment:
I suggest the following then, which will only re-run the query if it's not loaded, but only notifies the listeners if the `RealmObject/accessor` is still attached
```java
if (RealmObject.isLoaded(proxy)) { // already completed, no need to re-run
    if (RealmObject.isValid(proxy)) { // has a valid pointer (not deleted)
        RealmLog.trace("[COMPLETED_ASYNC_REALM_OBJECT %s], realm: %s. " +
                        "RealmObject is already loaded, just notify it",
                realm, HandlerController.this);
        proxy.realmGet$proxyState().notifyChangeListeners$realm();
    }

} else {
//....
```